compress.hh

```diff
@@ -20,8 +20,9 @@
 
 class compressor {
     sstring _name;
+    bool _writes_uncompressed_size;
```

Comment:
Ok, so this describes the underlying compressor, and this wrapper behaves in the opposite way (if the underlying compressor doesn't, we do)

Reply:
Deserves a comment IMO.

Reply:
Because reading the code is quite confusing - we check that `!_writes_uncompressed_size` and then we write uncompressed size...